src/popup/pages/Paratime/index.js

```diff
@@ -1,32 +1,45 @@
+import cx from "classnames";
 import React from "react";
 import { connect } from "react-redux";
 import loadingCommon from "../../../assets/images/loadingCommon.gif";
 import noHistory from "../../../assets/images/noHistory.png";
-import whiteArrow from "../../../assets/images/whiteArrow.svg";
+import wallet_send from "../../../assets/images/wallet_send.png";
+import sendDisable from "../../../assets/images/wallet_send_disable.svg";
 import { getRpcBalance, getRpcRuntimeList, getRuntimeNameList } from "../../../background/api";
-import { SEND_PAGE_TYPE_RUNTIME_DEPOSIT } from "../../../constant/types";
+import { SEND_PAGE_TYPE_RUNTIME_DEPOSIT, SEND_PAGE_TYPE_RUNTIME_WITHDRAW } from "../../../constant/types";
 import { ACCOUNT_TYPE } from "../../../constant/walletType";
 import { getLanguage } from "../../../i18n";
-import { updateRuntimeList, updateRuntimeNameList } from "../../../reducers/accountReducer";
+import { updateAccountLoading, updateRuntimeList, updateRuntimeNameList } from "../../../reducers/accountReducer";
 import { updateSendPageType } from "../../../reducers/cache";
 import { addressSlice, getRuntimeAddress } from "../../../utils/utils";
+import Button from "../../component/Button";
 import Clock from "../../component/Clock";
 import Toast from "../../component/Toast";
+import WalletBar from "../../component/WalletBar";
 import "./index.scss";
-
 class Paratime extends React.Component {
     constructor(props) {
         super(props);
         this.state = {
             refreshing: false,
-            loading: props.runtimeList.length === 0
+            loading: props.runtimeList.length === 0,
+            currentShowList: this.getShowRuntimeList()
         }
         this.isUnMounted = false;
         this.isRequest = false
     }
     async componentDidMount() {
         this.fetchData()
     }
+
+    getShowRuntimeList = () => {
```

Comment:
I sometimes get "unknown" as runtime names
![Peek 2021-12-10 04-21](https://user-images.githubusercontent.com/3758846/145513016-c0cc18e4-718f-476d-b060-2b64f9c9652c.gif)

Reply:
The runtime names returned by the network are used here. So there will be this situation, because grpc will continue to request runtime-names after returning to the list. When the runtime-names request is returned, it will be matched to display the correct name instead of the default unknown.
The purpose of using the interface to return the name here is to get the latest name in time 

https://github.com/oasisprotocol/oasis-wallet-ext/blob/121c3be81798b605bf548b9ec8893202e7ff5c81/src/popup/pages/Paratime/index.js#L94

Reply:
It looks like you need either
```diff
         getRuntimeNameList().then((data) => {
             this.props.updateRuntimeNameList(data)
+            this.callSetState({
+                currentShowList: this.getShowRuntimeList()
+            })
         })
```

or
```diff
-  getRuntimeNameList().then((data) => {
-      this.props.updateRuntimeNameList(data)
-  })
+  let names = getRuntimeNameList()
   let rpcBalance = getRpcBalance(currentAccount.address)
   let runtimeRequestList = getRpcRuntimeList()
-  Promise.all([rpcBalance, runtimeRequestList]).then(async (data) => {
+  Promise.all([rpcBalance, runtimeRequestList, names]).then(async (data) => {
+      this.props.updateRuntimeNameList(data[2])
       let rpcAccount = data[0]
```

Reply:
i change name and runtimeId match from net-data to local-data . It’s faster and won’t go wrong